### PR TITLE
Add SUDO_PROJECT_ID for openstack_dns_recordset_v2

### PR DESCRIPTION
**Pull request by sebastienmusso**
Add the capability to change recordset project ID for openstack_dns_recordset_v2

### CODE REVIEW

File: openstack/dns_zone_v2.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func dnsClientSetAuthHeader(resourceData *schema.ResourceData, dnsClient *gopher
 
 	if len(headers) != 0 {
 		dnsClient.MoreHeaders = headers
-		log.Printf("[DEBUG] openstack_dns_zone_v2 request headers set: %#v", headers)
+		log.Printf("[DEBUG] request headers set: %#v", headers)
```
Comment (ozerovandrei) [inline openstack/dns_zone_v2.go:73]: Why this message is changed?

File: openstack/resource_openstack_dns_recordset_v2.go [hunk 2]
```diff
@@ -116,6 +127,7 @@ func resourceDNSRecordSetV2Create(d *schema.ResourceData, meta interface{}) erro
 
 	zoneID := d.Get("zone_id").(string)
 	n, err := recordsets.Create(dnsClient, zoneID, createOpts).Extract()
+
```
Comment (ozerovandrei) [inline openstack/resource_openstack_dns_recordset_v2.go:130]: I don't think that separating `err` from `err != nil` is a good style.

File: openstack/resource_openstack_dns_recordset_v2_test.go [hunk 3]
```diff
@@ -2,6 +2,7 @@ package openstack
 
 import (
 	"fmt"
+	//"log"
```
Comment (ozerovandrei) [inline openstack/resource_openstack_dns_recordset_v2_test.go:5]: This should be removed.

File: openstack/resource_openstack_dns_recordset_v2_test.go [hunk 4]
```diff
@@ -181,13 +212,18 @@ func testAccCheckDNSV2RecordSetExists(n string, recordset *recordsets.RecordSet)
 		if rs.Primary.ID == "" {
 			return fmt.Errorf("No ID is set")
 		}
+		//log.Printf("[DEBUG] 2 ----------------------- %#v", rs.Primary.Attributes["project_id"])
```
Comment (ozerovandrei) [inline openstack/resource_openstack_dns_recordset_v2_test.go:215]: This should be removed.

**Review by ozerovandrei — CHANGES_REQUESTED** [review]

File: openstack/dns_zone_v2.go (hunk 1 above)
Comment (sebastienmusso) [reply openstack/dns_zone_v2.go:73]: Besause same function is used by resource of type recordset, in order to not being specific to dns zone resources only

File: openstack/resource_openstack_dns_recordset_v2_test.go (hunk 3 above)
Comment (sebastienmusso) [reply openstack/resource_openstack_dns_recordset_v2_test.go]: Fixed

File: openstack/resource_openstack_dns_recordset_v2_test.go (hunk 4 above)
Comment (sebastienmusso) [reply openstack/resource_openstack_dns_recordset_v2_test.go]: Fixed

File: openstack/resource_openstack_dns_recordset_v2.go (hunk 2 above)
Comment (sebastienmusso) [reply openstack/resource_openstack_dns_recordset_v2.go:130]: You mean have same error message for both dns client configuration ans dns header ?

Comment (ozerovandrei) [reply openstack/resource_openstack_dns_recordset_v2.go:130]: No I meant the empty line. Same error message is fine :)

Comment (sebastienmusso) [reply openstack/resource_openstack_dns_recordset_v2.go]: Yes right done in the last commit

**Comment by ozerovandrei** [comment]
Looks fine but I want to run `recheck designate` in this PR after https://github.com/terraform-provider-openstack/terraform-provider-openstack/issues/1253 is resolved.

**Comment by nikParasyr** [comment]
recheck

**Comment by nikParasyr** [comment]
recheck designate

**Comment by sebastienmusso** [comment]
> Build succeeded.
> 
> - [terraform-provider-openstack-unittest-foo ](https://logs.openlabtesting.org/logs/54/1254/651571abb8abeadf19a67af12d57227ec52cb0e2/check/terraform-provider-openstack-unittest-foo/b82bd9b/) : SUCCESS in 4m 19s
> - [terraform-provider-openstack-acceptance-test-foo ](https://logs.openlabtesting.org/logs/54/1254/651571abb8abeadf19a67af12d57227ec52cb0e2/check/terraform-provider-openstack-acceptance-test-foo/f2f848c/) : SUCCESS in 2h 15m 18s
> - [terraform-provider-openstack-acceptance-test-admin-foo ](https://logs.openlabtesting.org/logs/54/1254/651571abb8abeadf19a67af12d57227ec52cb0e2/check/terraform-provider-openstack-acceptance-test-admin-foo/e931b38/) : SUCCESS in 49m 45s
> 

Yes !!! Here we go

**Comment by nikParasyr** [comment]
> Build failed.
> 
>     * [terraform-provider-openstack-acceptance-test-designate-foo ](https://logs.openlabtesting.org/logs/54/1254/651571abb8abeadf19a67af12d57227ec52cb0e2/recheck-designate/terraform-provider-openstack-acceptance-test-designate-foo/770c676/) : FAILURE in 39m 38s

@sebastienmusso This test-set is more important and it is failing. I haven't had the time to go through the logs etc but by searching for `FAIL` you should be able to see which tests fail.

**Comment by sebastienmusso** [comment]
Hi,

it seems to miss some permission within openstack api:

```
POST http://192.168.0.21/identity/v3/projects], error message: {"error":{"code":403,"message":"You are not authorized to perform the requested action: identity:create_project.","title":"Forbidden"}}
```

In the test, I create an openstack project and a dns record inside it.
The test user does't seems to have sufficient permissions.

**Comment by nikParasyr** [comment]
@sebastienmusso sorry for the late reply. After having a look, it is a bit harder to properly fix the test scenario. I'll give some context:

In general a job(and all tests in that job) run as a single openstack user, who is either a `member` (the typical openstack `user` role) or an `admin`. Some tests need to be run on a job that is admin and some on the one that is a member. Unfortunately currently we don't have a more dynamic way of changing the permissions of a user within a job.  For that reason our typical check has 2 jobs(excluding the unit tests), one for the admin tests and one for the user tests.

The designate job runs as a `member`, which means you don't have permissions to create a project (which is required for your test case). Some of the options:
1. You don't create a project but you get the current project_id that the user belongs to (basically the env var OS_PROJECT_ID, a helper function could do that and then you pass it as a var in your`testAccDNSV2RecordSetDifferentProject`) and use that (i'm not 100% sure this will be enough).
2. We create a new job for designate where you have admin credentials therefore you can create a project. In this case and based on your docs(`The ID of the project DNS zone is created
  for, sets X-Auth-Sudo-Tenant-ID header (requires an assigned 
  user role in target project)`) you should also add a role association to that project which is currently missing.
  
  
My opinion is that if case 1 works then that's better cause we don't add yet another job. Otherwise for the option 2 i can create an MR with the new admin job, unfortunately it might take some time for our test infrastructure to be updated due to how its being managed. After that you can rebase and update your test accordingly. I'd like @ozerovandrei opinion on this. It's quite some work to cover one test case unfortunately.

Hopefully this explains some things to you.

**Comment by ozerovandrei** [comment]
It seems that the first way is easier if it worked.

**Comment by sebastienmusso** [comment]
I simplify test scenario. No more project are created and dns entry is created with "project_id" argument

**Comment by nikParasyr** [comment]
recheck designate

**Comment by sebastienmusso** [comment]
recheck designate

**Comment by sebastienmusso** [comment]
Here we go !

**Comment by sebastienmusso** [comment]
recheck designate

**Review by ozerovandrei — APPROVED** [review]
LGTM

Thank you @sebastienmusso

**Pull request merged**